File: SWP.Core/Entities/Vehicle.cs
using SWP.Core.GMSAttribute;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SWP.Core.Entities
{
    [GMSTableName("vehicle")]
    public class Vehicle
    {
        [Column("vehicle_id")]
        [Key]
        public int VehicleId { get; set; }

        [Column("vehicle_name")]
        [Required]
        [MaxLength(100)]
        public string VehicleName { get; set; } = string.Empty;

        [Column("vehicle_license_plate")]
        [Required]
        [MaxLength(255)]
        public string VehicleLicensePlate { get; set; } = string.Empty;

        [Column("current_km")]
        public int? CurrentKm { get; set; }

        [Column("customer_id")]
        public int? CustomerId { get; set; }

        [Column("make")]
        [MaxLength(255)]
        public string? Make { get; set; }

        [Column("model")]
        [MaxLength(255)]
        public string? Model { get; set; }

        [Column("is_deleted")]
        public int IsDeleted { get; set; } = 0;

        // Navigation properties
        public Customer? Customer { get; set; }
        public ICollection<ServiceTicket> ServiceTickets { get; set; } = new List<ServiceTicket>();
    }
}

File: GarageProject-BE-Springboot-MainFlow/src/main/java/fptu/edu/vn/training/entity/Vehicle.java
package fptu.edu.vn.training.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.LinkedHashSet;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "vehicle")
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "vehicle_id")
    private Integer vehicleId;

    @Column(name = "vehicle_name", nullable = false, length = 100)
    private String vehicleName;

    @Column(name = "vehicle_license_plate", nullable = false, length = 255)
    private String vehicleLicensePlate;

    @Column(name = "current_km")
    private Integer currentKm;

    @Column(name = "customer_id")
    private Integer customerId;

    @Column(name = "make", length = 255)
    private String make;

    @Column(name = "model", length = 255)
    private String model;

    @Column(name = "is_deleted", nullable = false)
    @Builder.Default
    private Integer isDeleted = 0;

    @Builder.Default
    @OneToMany(mappedBy = "vehicle", fetch = FetchType.LAZY, cascade = CascadeType.ALL, orphanRemoval = false)
    private Set<ServiceTicket> serviceTickets = new LinkedHashSet<>();
}
